acquia_lift.batch.inc

```diff
@@ -112,12 +149,53 @@ function acquia_lift_batch_finished($success, $results, $operations) {
     // $operations contains the operations that remained unprocessed.
     $error_operation = reset($operations);
     $message = t('An error occurred while processing %error_operation with arguments: @arguments', array('%error_operation' => $error_operation[0], '@arguments' => print_r($error_operation[1], TRUE)));
-    drupal_set_message($message, 'error');
+    _acquia_lift_batch_set_error($message);
   }
 }
 
+/**
+ * Sets an error in a session variable indicating failure during batch sync.
+ *
+ * @param $error
+ *   The error message to store.
+ */
+function _acquia_lift_batch_set_error($error) {
+  $_SESSION['acquia_lift_batch_error'] = $error;
+}
+
+/**
+ * Returns any error that was set during batch syncing.
+ *
+ * Also unsets the session variable used to track errors. This function should
+ * only ever be called by acquia_lift_batch_sync_tests_for_agent().
+ *
+ * @return string|NULL
+ *   The most recently set error or NULL if none was set
+ */
+function _acquia_lift_batch_get_error() {
```

Comment:
Is there a reason to only hold on to the most recent error rather than all errors?